From cbec33705156cb58fa65fd2c3346c46a605428ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Wed, 31 Jul 2019 23:24:41 +0200
Subject: [PATCH] build: Drop intltool

Any non-ancient version of gettext supports the functionality
previously provided by intltool.
---
 autogen.sh                                    |  1 -
 configure.ac                                  |  3 +-
 data/Makefile.am                              |  8 +--
 ....freedesktop.Telepathy.Logger.gschema.xml} |  6 +-
 po/Makevars                                   | 68 +++++++++++++++++++
 po/POTFILES.in                                |  2 +-
 6 files changed, 75 insertions(+), 13 deletions(-)
 rename data/{org.freedesktop.Telepathy.Logger.gschema.xml.in => org.freedesktop.Telepathy.Logger.gschema.xml} (79%)
 create mode 100644 po/Makevars

diff --git a/autogen.sh b/autogen.sh
index 58ee191..0d8e27d 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,7 +1,6 @@
 #!/bin/sh
 set -e
 
-intltoolize --force --copy --automake || exit 1
 gtkdocize || exit 1
 
 if test -n "$AUTOMAKE"; then
diff --git a/configure.ac b/configure.ac
index 332e8ce..6391678 100644
--- a/configure.ac
+++ b/configure.ac
@@ -95,7 +95,8 @@ LT_INIT
 AM_PROG_MKDIR_P
 AM_PATH_GLIB_2_0
 AC_PATH_XTRA
-IT_PROG_INTLTOOL([0.35.0])
+AM_GNU_GETTEXT_VERSION([0.19.6])
+AM_GNU_GETTEXT([external])
 AC_PATH_PROG(DBUS_BINDING_TOOL, dbus-binding-tool)
 GLIB_GSETTINGS
 GLIB_GENMARSHAL=`$PKG_CONFIG glib-2.0 --variable=glib_genmarshal`
diff --git a/data/Makefile.am b/data/Makefile.am
index 1e2d780..f6949f7 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -1,8 +1,4 @@
-gsettings_files = \
-	org.freedesktop.Telepathy.Logger.gschema.xml.in \
-	$(NULL)
-gsettings_SCHEMAS = $(gsettings_files:.xml.in=.xml)
-@INTLTOOL_XML_NOMERGE_RULE@
+gsettings_SCHEMAS = org.freedesktop.Telepathy.Logger.gschema.xml
 @GSETTINGS_RULES@
 
 servicefiledir = $(datadir)/dbus-1/services
@@ -23,13 +19,11 @@ clientfile_parts = $(top_srcdir)/data/Logger.Observer.client.in \
 clientfile_DATA = Logger.client
 
 EXTRA_DIST = 				\
-	$(gsettings_files)		\
 	$(clientfile_DATA)		\
 	$(servicefile_in_files)		\
 	$(clientfile_parts)
 
 CLEANFILES = \
-	$(gsettings_SCHEMAS) \
 	$(servicefile_DATA) \
 	$(clientfile_DATA)
 
diff --git a/data/org.freedesktop.Telepathy.Logger.gschema.xml.in b/data/org.freedesktop.Telepathy.Logger.gschema.xml
similarity index 79%
rename from data/org.freedesktop.Telepathy.Logger.gschema.xml.in
rename to data/org.freedesktop.Telepathy.Logger.gschema.xml
index ddb4c15..76d145d 100644
--- a/data/org.freedesktop.Telepathy.Logger.gschema.xml.in
+++ b/data/org.freedesktop.Telepathy.Logger.gschema.xml
@@ -2,11 +2,11 @@
   <schema id="org.freedesktop.Telepathy.Logger" path="/apps/telepathy-logger/">
     <key name="enabled" type="b">
       <default>true</default>
-      <_summary>Enable logging</_summary>
-      <_description>
+      <summary>Enable logging</summary>
+      <description>
         Globally enable or disable the Telepathy logger system. Setting it to
         "false" will completely disable all logging.
-      </_description>
+      </description>
     </key>
   </schema>
 </schemalist>
diff --git a/po/Makevars b/po/Makevars
new file mode 100644
index 0000000..30799ce
--- /dev/null
+++ b/po/Makevars
@@ -0,0 +1,68 @@
+# Makefile variables for PO directory in any package using GNU gettext.
+
+# Usually the message domain is the same as the package name.
+DOMAIN = $(PACKAGE)
+
+# These two variables depend on the location of this directory.
+subdir = po
+top_builddir = ..
+
+# These options get passed to xgettext.
+XGETTEXT_OPTIONS = --from-code=UTF-8 --keyword=_ --keyword=N_ \
+        --keyword=C_:1c,2 --keyword=NC_:1c,2 \
+        --keyword=g_dngettext:2,3 --add-comments \
+        --flag=g_dngettext:2:pass-c-format \
+        --flag=g_strdup_printf:1:c-format \
+        --flag=g_string_printf:2:c-format \
+        --flag=g_string_append_printf:2:c-format \
+        --flag=g_error_new:3:c-format \
+        --flag=g_set_error:4:c-format \
+        --flag=g_markup_printf_escaped:1:c-format \
+        --flag=g_log:3:c-format \
+        --flag=g_print:1:c-format \
+        --flag=g_printerr:1:c-format \
+        --flag=g_printf:1:c-format \
+        --flag=g_fprintf:2:c-format \
+        --flag=g_sprintf:2:c-format \
+        --flag=g_snprintf:3:c-format
+
+
+# This is the copyright holder that gets inserted into the header of the
+# $(DOMAIN).pot file.  Set this to the copyright holder of the surrounding
+# package.  (Note that the msgstr strings, extracted from the package's
+# sources, belong to the copyright holder of the package.)  Translators are
+# expected to transfer the copyright for their translations to this person
+# or entity, or to disclaim their copyright.  The empty string stands for
+# the public domain; in this case the translators are expected to disclaim
+# their copyright.
+COPYRIGHT_HOLDER = Translation copyright holder
+# This is the email address or URL to which the translators shall report
+# bugs in the untranslated strings:
+# - Strings which are not entire sentences, see the maintainer guidelines
+#   in the GNU gettext documentation, section 'Preparing Strings'.
+# - Strings which use unclear terms or require additional context to be
+#   understood.
+# - Strings which make invalid assumptions about notation of date, time or
+#   money.
+# - Pluralisation problems.
+# - Incorrect English spelling.
+# - Incorrect formatting.
+# It can be your email address, or a mailing list address where translators
+# can write to without being subscribed, or the URL of a web page through
+# which the translators can contact you.
+MSGID_BUGS_ADDRESS = http://bugzilla.gnome.org/enter_bug.cgi?product=polari&keywords=I18N+L10N&component=general
+
+# This is the list of locale categories, beyond LC_MESSAGES, for which the
+# message catalogs shall be used.  It is usually empty.
+EXTRA_LOCALE_CATEGORIES =
+
+# Ignore the timestamp of the .pot file, as git clones do not have
+# deterministic timestamps, and .po files are updated by translators
+# (only) in GNOME projects.
+PO_DEPENDS_ON_POT = no
+
+# This tells whether or not to forcibly update $(DOMAIN).pot and
+# regenerate PO files on "make dist".  Possible values are "yes" and
+# "no".  Set this to no if the POT file and PO files are maintained
+# externally.
+DIST_DEPENDS_ON_UPDATE_PO = no
diff --git a/po/POTFILES.in b/po/POTFILES.in
index 300afaf..69c5b7c 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -1 +1 @@
-data/org.freedesktop.Telepathy.Logger.gschema.xml.in
+data/org.freedesktop.Telepathy.Logger.gschema.xml
-- 
2.21.0

